File: app/src/androidTest/java/com/android/example/sunshine/app/data/TestWeatherContract.java
package com.android.example.sunshine.app.data;

import android.net.Uri;
import android.test.AndroidTestCase;

public class TestWeatherContract extends AndroidTestCase {

    // intentionally includes a slash to make sure Uri is getting quoted correctly
    private static final String TEST_WEATHER_LOCATION = "/North Pole";
    private static final long TEST_WEATHER_DATE = 1419033600L;  // December 20th, 2014

    public void testBuildWeatherLocation() {
        Uri locationUri = WeatherContract.WeatherEntry.buildWeatherLocation(TEST_WEATHER_LOCATION);
        assertNotNull("Null Uri returned.  You must fill-in buildWeatherLocation in WeatherContract.",
                locationUri);
        assertEquals("Weather location should be appended to the end of the Uri",
                TEST_WEATHER_LOCATION, locationUri.getLastPathSegment());
        assertEquals("Error: Weather location Uri doesn't match our expected result",
                locationUri.toString(), "content://com.android.example.sunshine.app/weather/%2FNorth%20Pole");
    }
}

File: app/src/androidTest/java/com/android/example/sunshine/app/data/TestWeatherContract.kt
package com.android.example.sunshine.app.data

import android.test.AndroidTestCase

class TestWeatherContract : AndroidTestCase() {

    fun testBuildWeatherLocation() {
        val locationUri = WeatherContract.WeatherEntry.buildWeatherLocation(TEST_WEATHER_LOCATION)
        assertNotNull("Null Uri returned.  You must fill-in buildWeatherLocation in WeatherContract.",
                locationUri)
        assertEquals("Weather location should be appended to the end of the Uri",
                TEST_WEATHER_LOCATION, locationUri.lastPathSegment)
        assertEquals("Error: Weather location Uri doesn't match our expected result",
                locationUri.toString(), "content://com.android.example.sunshine.app/weather/%2FNorth%20Pole")
    }

    companion object {
        // intentionally includes a slash to make sure Uri is getting quoted correctly
        val TEST_WEATHER_LOCATION = "/North Pole"
    }
}
